backend/routes/recipe.js

```diff
@@ -0,0 +1,75 @@
+const Recipe = require("../models/recipeSchema")
+const express = require('express')
+
+const router = express.Router()
+
+router.get("/", async (req, res) => {
+    const recipe = await Recipe.find({})
+    res.send(recipe)
+})
+
+router.get('/:recipeName', async (req, res) => {
+    const name = req.params.recipeName
+    const recipe = await Recipe.findOne({
+        'recipeName': name
+    });
+    res.send(recipeName)
+})
+
+router.post("/", async (req, res) => {
+    const { recipeName, recipeDescription, recipeInfo, recipeImage, ingredientList, steps, link } = req.body
+    let recipe = new Recipe({
+        recipeName,
+        recipeDescription,
+        recipeInfo,
+        recipeImage,
+        ingredientList,
+        steps,
+        link
+    });
+    try {
+        recipe = await recipe.save();
+        res.send("${recipeName} added to collection");
+    }
+    catch (error) {
+        res.status(500).send(error.message);
```

Comment:
rather than using 500 error, we can return error.code to display the appropriate error